test/functional/legacy/assert_spec.lua

```diff
@@ -142,4 +142,16 @@ describe('assert function:', function()
       })
     end)
   end)
+
+  it('should allow setting to a number', function()
```

Comment:
This is actually testing whether the old value of `v:errors` is properly freed when an `assert_*` function sets a new value, regardless of the old type of `v:errors` (which is supposed to just be a list).

Maybe reword this to `it('is reset to a list by assert functions', function()`?